consensus/src/network_interface.rs

```diff
@@ -104,27 +106,41 @@ impl NewNetworkSender for ConsensusNetworkSender {
 }
 
 impl ConsensusNetworkSender {
-    /// Send a single message to the destination peer using the `CONSENSUS_DIRECT_SEND_PROTOCOL`
-    /// ProtocolId.
+    /// Send a single message to the destination peer using available ProtocolId.
     pub async fn send_to(
         &mut self,
         recipient: PeerId,
         message: ConsensusMsg,
     ) -> Result<(), NetworkError> {
-        let protocol = ProtocolId::ConsensusDirectSend;
-        self.network_sender.send_to(recipient, protocol, message)
+        for protocol in DIRECT_SEND {
+            if self.network_sender.is_available(recipient, *protocol).await {
```

Comment:
yeah this will probably be very slow : p

Reply:
it's actually fine in cluster test, there's no difference compared to the serde/network latency.

the double serde in send_to_many adds ~10ms latency

Reply:
but I'll use NewPeerEvent anyway since it looks nicer